File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/connection/IdentificationFailedBannedMessage.scala





















// Generated on 05/03/2013 10:34:31
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait IdentificationFailedBannedMessage extends IdentificationFailedMessage {
  override val messageId = 6174

  protected trait Message extends IdentificationFailedMessage.Message with Serializable {
    override val messageId = IdentificationFailedBannedMessage.messageId

    val banEndDate: Double

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeDouble(banEndDate)
    }

  }

  protected def create(reason: Byte, banEndDate: Double): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val reason = buf.readByte()
    val banEndDate = buf.readDouble()

    create(reason, banEndDate)
  }

}

object IdentificationFailedBannedMessage extends IdentificationFailedBannedMessage {
  case class MessageImpl(reason: Byte, banEndDate: Double) extends Message
  type Target = MessageImpl

  override def create(reason: Byte, banEndDate: Double) = MessageImpl(reason, banEndDate)

  def apply(reason: Byte, banEndDate: Double) = create(reason, banEndDate)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/IdentificationFailedBannedMessage.java


// Generated on 05/08/2013 19:37:37
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class IdentificationFailedBannedMessage extends IdentificationFailedMessage {
    public static final int MESSAGE_ID = 6174;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public double banEndDate;
    
    public IdentificationFailedBannedMessage() { }
    
    public IdentificationFailedBannedMessage(byte reason, double banEndDate) {
        super(reason);
        this.banEndDate = banEndDate;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeDouble(banEndDate);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        banEndDate = buf.readDouble();
        if (banEndDate < 0)
            throw new RuntimeException("Forbidden value on banEndDate = " + banEndDate + ", it doesn't respect the following condition : banEndDate < 0");
    }
    
}
